### PR TITLE
feat: Data source implementation and tests for `mongodbatlas_employee_access_grant`

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Data source implementation and tests for `mongodbatlas_employee_access_grant`.  

Also tests are run in CI.

Link to any related issue(s): CLOUDP-272252

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: templates/data-sources/employee_access_grant.tmpl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# {{.Type}}: {{.Name}}
```
Comment (lantoli) [inline templates/data-sources/employee_access_grant.tmpl:1]: min doc so code-health GHA doesn't fail

File: templates/data-sources/employee_access_grant.md.tmpl [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# {{.Type}}: {{.Name}}
```
Comment (lantoli) [inline templates/data-sources/employee_access_grant.md.tmpl:1]: min doc so code-health GHA doesn't fail

File: internal/service/employeeaccessgrant/data_source.go [hunk 3]
```diff
@@ -27,4 +27,23 @@ func (d *employeeAccessGrantDS) Schema(ctx context.Context, req datasource.Schem
...
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	connV2 := d.Client.AtlasV2
+	projectID := tfModel.ProjectID.ValueString()
+	clusterName := tfModel.ClusterName.ValueString()
+	cluster, _, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
+	atlasResp, _ := cluster.GetMongoDBEmployeeAccessGrantOk()
+	if err != nil || atlasResp == nil {
+		msg := "info not found"
```
Comment (EspenAlbert) [inline internal/service/employeeaccessgrant/data_source.go]: can this msg be more specific?

**Review by EspenAlbert — APPROVED** [review]
LGTM!

File: internal/service/employeeaccessgrant/data_source.go (hunk 3 above)
Comment (lantoli) [reply internal/service/employeeaccessgrant/data_source.go]: thanks, done in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2580/commits/943775a8a9badd9d78fe47cd1d1b0cc6a1ac44da

**Pull request merged**